scripting/get5/recording.sp

```diff
@@ -137,3 +226,35 @@ void SetCurrentMatchRestartDelay(float delay) {
     mp_match_restart_delay.FloatValue = delay;
   }
 }
+
+static int DemoRequestCallback(Handle request, bool failure, bool requestSuccessful,
```

Comment:
Make the callback function this. Note that if you rename `get5_demo_upload_delete_after_upload`, also replace it here. I tested this code and it works for success, HTTP errors and network errors.

```c
static int DemoRequestCallback(Handle request, bool failure, bool requestSuccessful,
                               EHTTPStatusCode statusCode) {
  if (failure || !requestSuccessful) {
    LogError("Failed to upload demo %s.", g_DemoFileNameLastStartedUpload);
    delete request;
    return;
  }
  int status = view_as<int>(statusCode);
  if (status >= 300 || status < 200) {
    LogError("Demo request failed with HTTP status code: %d.", statusCode);
    int responseSize;
    SteamWorks_GetHTTPResponseBodySize(request, responseSize);
    char[] response = new char[responseSize];
    if (SteamWorks_GetHTTPResponseBodyData(request, response, responseSize)) {
      LogError("Response body: %s", response);
    } else {
      LogError("Failed to read response body.");
    }
    delete request;
    return;
  }
  LogDebug("Demo request succeeded. HTTP status code: %d.", statusCode);
  if (g_DemoUploadDeleteAfterCvar.BoolValue && !StrEqual(g_DemoFileNameLastStartedUpload, "")) {
    LogDebug("get5_demo_upload_delete_after_upload set to true; deleting the file from the game server.");
    if (FileExists(g_DemoFileNameLastStartedUpload)) {
      if (!DeleteFile(g_DemoFileNameLastStartedUpload)) {
        LogError("Unable to delete demo file %s.", g_DemoFileNameLastStartedUpload);
      }
    }
  }
  delete request;
}
```